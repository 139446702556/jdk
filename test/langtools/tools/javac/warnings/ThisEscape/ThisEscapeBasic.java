/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @compile/ref=ThisEscapeBasic.out -Xlint:this-escape -XDrawDiagnostics ThisEscapeBasic.java
 * @summary Verify basic 'this' escape detection
 */

public class ThisEscapeBasic {

    public ThisEscapeBasic() {
        this.mightLeak();
    }

    public void mightLeak() {
    }
}
