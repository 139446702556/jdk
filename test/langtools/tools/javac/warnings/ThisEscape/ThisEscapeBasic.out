ThisEscapeBasic.java:11:23: compiler.warn.possible.this.escape
1 warning
